spring-xd-dirt/src/main/java/org/springframework/xd/dirt/module/ExtendedResource.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.xd.dirt.module;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+import org.springframework.util.FileSystemUtils;
+import org.springframework.util.ReflectionUtils;
+
+/**
+ * A wrapper around a Resource that adds capabilities lacking in the original interface.
+ *
+ * @since 1.2
+ * @author Eric Bottard
+ */
+public abstract class ExtendedResource {
+
+	private static final Class<?> HDFS_RESOURCE;
+
+	static {
+		try {
+			HDFS_RESOURCE = Class.forName("org.springframework.data.hadoop.fs.HdfsResource");
+		}
+		catch (ClassNotFoundException e) {
+			throw new IllegalStateException("Spring Data Hadoop not found on the classpath, but was expected", e);
```

Comment:
One way to do it would be override the bean definition(s) for the registry(ies) in the test.

I wonder if removing the hard class dependency (using classnames) would be enough though